conda/common/io.py

```diff
@@ -689,6 +693,114 @@ def _ensure_dir(self):
             os.makedirs(dirname(self.record_file))
 
 
+Reporter = Dict[str, Union[bool, int, str]]
+DetailRecord = Dict[str, Union[str, int, bool]]
+
+
+class ReporterHandlerBase(ABC):
+    """
+    Base class for all reporter handlers
+    """
+
+    @abstractmethod
+    def detail_view(self, data: DetailRecord, **kwargs) -> str: ...
+
+    @abstractmethod
+    def string_view(self, data: str, **kwargs) -> str: ...
+
+
+class ConsoleHandler(ReporterHandlerBase):
+    def detail_view(self, data: DetailRecord, **kwargs) -> str:
+        table_str = ""
+        longest_header = len(sorted(data.keys(), key=len).pop())
+
+        for header, value in data.items():
+            row_header = header.ljust(longest_header, " ")
+            table_str += f"{row_header} : {value}\n"
+
+        return table_str
+
+    def string_view(self, data: str, **kwargs) -> str:
+        return data
+
+
+class JSONHandler(ReporterHandlerBase):
```

Comment:
It would be nice to have a comment here as well to briefly describe how this behaves.